### ISSUE TITLE
.spz support

### ISSUE THREAD

**Issue by cyango** [mkkellogg/GaussianSplats3D#381]
As scanniverse released their open source new .spz which seems to bring new optimizations, is this library capable of supporting that .spz?

https://scaniverse.com/news/spz-gaussian-splat-open-source-file-format

**Comment by seppestaes** [comment]
Ongoing discussion #47

**Comment by mkkellogg** [comment]
I have a branch in development that supports the `.spz` format (up to 2nd degree spherical harmonics) as well as the updated PlayCanvas compression with spherical harmonics (also up to 2nd degree): https://github.com/mkkellogg/GaussianSplats3D/tree/playcanvas_and_spz_compression. Anyone want to test it out?

**Comment by seppestaes** [comment]
Hi,

I tested the updated PlayCanvas compression.

- SH0 & SH2 look great
- SH1 gives color distortions here:

<img width="580" alt="Screenshot 2024-12-29 at 12 12 22" src="https://github.com/user-attachments/assets/7cb37e0a-f537-4f24-bda4-39632e1aed6a" />

Talking about supported input formats :-)
Would it be possible to support SH2 for Nerfstudio's .ply?
The field names correspond to INRIA2's with the exception of 'f_rest_45'.
I changed the sectionName from 'codebook_centers' to 'vertex' both in INRIAV2PlyParser.js and PlyParserUtils.js for testing purposes. No parsing errors, but I do get a blank (black) screen.

Merry Christmas! :-)

`const FieldNamesToRead = ['scale_0', 'scale_1', 'scale_2', 'rot_0', 'rot_1', 'rot_2', 'rot_3',
 'x', 'y', 'z', 'f_dc_0', 'f_dc_1', 'f_dc_2', 'opacity', 'red', 'green', 'blue',
 'f_rest_0', 'f_rest_1', 'f_rest_2', 'f_rest_3', 'f_rest_4', 'f_rest_5', 'f_rest_6', 'f_rest_7', 'f_rest_8',
'f_rest_9', 'f_rest_10', 'f_rest_11', 'f_rest_12', 'f_rest_13', 'f_rest_14', 'f_rest_15', 'f_rest_16',
'f_rest_17', 'f_rest_18', 'f_rest_19', 'f_rest_20', 'f_rest_21', 'f_rest_22', 'f_rest_23', 'f_rest_24',
'f_rest_25', 'f_rest_26', 'f_rest_27', 'f_rest_28', 'f_rest_29', 'f_rest_30', 'f_rest_31', 'f_rest_32',
'f_rest_33', 'f_rest_34', 'f_rest_35', 'f_rest_36', 'f_rest_37', 'f_rest_38', 'f_rest_39', 'f_rest_40',
'f_rest_41', 'f_rest_42', 'f_rest_43', 'f_rest_44', 'f_rest_45'
];`


Nerfstudio PLY  

`ply
format binary_little_endian 1.0
comment Created by Open3D
element vertex 908403
property float x
property float y
property float z
property float nx
property float ny
property float nz
property float rot_2
property float scale_2
property float scale_0
property float opacity
property float f_rest_43
property float f_rest_42
property float f_rest_39
property float f_rest_36
property float f_rest_35
property float f_rest_33
property float f_rest_31
property float f_rest_34
property float f_rest_30
property float f_rest_29
property float rot_1
property float f_rest_28
property float scale_1
property float f_rest_38
property float f_rest_27
property float f_rest_40
property float f_rest_26
property float f_rest_41
property float f_rest_25
property float f_rest_23
property float f_rest_22
property float f_rest_21
property float f_rest_19
property float f_rest_16
property float f_rest_7
property float f_rest_1
property float f_rest_24
property float f_rest_13
property float f_rest_0
property float f_dc_2
property float f_rest_2
property float f_rest_5
property float f_dc_0
property float f_rest_17
property float f_rest_10
property float f_rest_12
property float f_rest_14
property float rot_3
property float f_rest_11
property float f_dc_1
property float f_rest_32
property float f_rest_18
property float f_rest_37
property float f_rest_6
property float f_rest_20
property float f_rest_3
property float f_rest_8
property float rot_0
property float f_rest_4
property float f_rest_44
property float f_rest_9
property float f_rest_15
end_header`

**Comment by mkkellogg** [comment]
Can you share the code you used to view the scene? I'm able to view the scene you shared just fine, so one of your viewer settings may be the culprit:
<img width="956" alt="image" src="https://github.com/user-attachments/assets/9b6f9818-f3f1-4fc6-970e-50a71675f038" />

**Comment by seppestaes** [comment]
Hi @mkkellogg,
I'm using the demo viewer (index.html). Added a block to FileBufferToSplatBuffer for the new SpzLoader. It's working nicely!

**Comment by cyango** [comment]
How much FPS are you getting? Does it load fast in mobile?

**Comment by KuoFengYuan** [comment]
Hi @mkkellogg,
I'm using the demo viewer open spz
but i use mobile phone safari open will crash
my url:
https://test.venraas.tw/demo.html?url=https://storage.googleapis.com/macaron_melody/will/1106_sofa.spz

**Comment by cyango** [comment]
I'm trying this on a RF3 component, but I'm having a error on import from that specific branch. Am i doing something wrong?

<img width="1127" alt="image" src="https://github.com/user-attachments/assets/7cb360e6-fba5-4bc4-901a-e28094b6796f" />

<img width="1002" alt="image" src="https://github.com/user-attachments/assets/406fe127-d472-4b50-9f66-0a327e1cdb22" />

**Comment by seppestaes** [comment]
@cyango : Did you build the package? 
`
"module": "build/gaussian-splats-3d.module.js",
"main": "build/gaussian-splats-3d.umd.cjs",
`

**Comment by cyango** [comment]
Yes I missed that step. Working on my R3F app! The only thing that is happening right now (but this is happening in the main branch too) is the react component is like re-rendering the splat everytime i click outside the browser window. Any reason why?

![image](https://github.com/user-attachments/assets/6aa560dd-8f40-496c-97dd-9612dd25bc18)

**Comment by cyango** [comment]
What's missing to merge that branch and publish to npm?

**Comment by mkkellogg** [comment]
SPZ support is now in the latest release: [v0.4.7](https://github.com/mkkellogg/GaussianSplats3D/releases/tag/v0.4.7)
